packages/botonic-react/src/util/webchat.js

```diff
@@ -0,0 +1,151 @@
+import { INPUT } from '@botonic/core'
+
+import { Button } from '../components/button'
+import { WEBCHAT } from '../constants'
+import { getProperty, strToBool } from './objects'
+import { deepMapWithIndex } from './react'
+
+/**
+ * Returns the value of a property defined in bot's theme based on WEBCHAT.CUSTOM_PROPERTIES dictionary.
+ * It gives preference to nested defined properties (e.g.: header.style) over plain properties (e.g.: headerStyle).
+ * If property doesn't exist, returns the defaultValue.
+ */
+export const _getThemeProperty = theme => (
+  property,
+  defaultValue = undefined
+) => {
+  for (const [k, v] of Object.entries(WEBCHAT.CUSTOM_PROPERTIES)) {
+    if (v == property) {
+      const nestedProperty = getProperty(theme, v)
+      if (nestedProperty !== undefined) return nestedProperty
+      const plainProperty = getProperty(theme, k)
+      if (plainProperty !== undefined) return plainProperty
+      return defaultValue
+    }
+  }
+  return undefined
+}
+
+export class ButtonsDisabler {
+  static constructBrowserProps(props) {
+    const disabledProps = { disabled: props.disabled }
+    if (props.autodisable !== undefined)
+      disabledProps.autodisable = strToBool(props.autodisable)
+    if (props.disabledstyle !== undefined)
+      disabledProps.disabledstyle = props.disabledstyle
+    return disabledProps
+  }
+  static constructNodeProps(props) {
+    const disabledProps = {}
+    if (props.autodisable !== undefined)
+      disabledProps.autodisable = String(props.autodisable)
+    if (props.disabledstyle !== undefined)
+      disabledProps.disabledstyle = JSON.stringify(props.disabledstyle)
+    return disabledProps
+  }
+
+  static withDisabledProps(props) {
+    return {
+      ...{ disabled: props.disabled },
+      ...{ autodisable: props.autodisable },
+      ...{ disabledstyle: props.disabledstyle },
+    }
```

Comment:
I see this all over, you don't need to create a new object to just deconstruct it. Simpler:

```
return {
  disabled: props.disabled,
  ...
}
```

Reply:
true, changed 👍